src/map/skill.cpp

```diff
@@ -7711,6 +7710,12 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 			return 1;
 		}
 		status_damage(src, src, sstatus->max_hp,0,0,1, skill_id);
+		if(skill_id == HVAN_EXPLOSION && src->type == BL_HOM) {
+			TBL_HOM *hd = BL_CAST(BL_HOM,src);
```

Comment:
Please use normal cast here.
The BL type was already checked above.
```suggestion
			struct homun_data *hd = (struct homun_data *)src;
```